### PR TITLE
Reference to enclosing element is not a usage

### PR DESCRIPTION
**Pull request by som-snytt**
In the linting arms race with dotty, fix an old `TODO`.

This will conflict with the PR which fixes message sorting.

Edit: it actually detected a few unused methods here.

### CODE REVIEW

**Comment by som-snytt** [comment]
Whaddaya mean, I didn't push the check file update? I forgot to `partest --real-easy`, which is real easy!

**Comment by som-snytt** [comment]
@SethTisue if you are available for a trivial review, I can reconcile the PR for linting for comprehensions.

**Review by SethTisue — APPROVED** [review]

File: src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala [hunk 1]
```diff
@@ -562,7 +562,7 @@ trait TypeDiagnostics extends splain.SplainDiagnostics {
             case b @ Bind(n, _) if !atBounded(b) && n != nme.DEFAULT_CASE => patvars += b.symbol
             case _ =>
           }
-        case _: RefTree if isExisting(sym)            => targets += sym
+        case _: RefTree => if (isExisting(sym) && !currentOwner.hasTransOwner(sym)) targets += sym
```
Comment (som-snytt) [inline src/compiler/scala/tools/nsc/typechecker/TypeDiagnostics.scala:565]: A follow-up improvement to check `!targets(sym)` before `hasTransOwner`.

**Pull request merged**